include/class.ostsession.php

```diff
@@ -200,7 +202,9 @@ function update($id, $data){
         $ttl = $this && method_exists($this, 'getTTL')
             ? $this->getTTL() : SESSION_TTL;
 
-        assert($this->data->session_id == $id);
+        // Create a session data obj if not loaded.
+        if (!$this->data)
+            $this->data = new SessionData(['session_id' => $id]);
```

Comment:
Looks like `session_rengenerate_id()` will also change the ID without calling any other functions. I wonder if we should write something like
```php
if (!$this->data)
    $this->data = new SessionData(['session_id' => $id])
elseif ($this->data->session_id != $id)
    $this->data->session_id = $id;
```